network/framework/src/connectivity_manager/mod.rs

```diff
@@ -436,47 +482,43 @@ where
     /// reconfiguration. If we are currently connected to this validator, calling
     /// this function will close our connection to it.
     async fn close_stale_connections(&mut self) {
```

Comment:
something is wrong with this implementation of 'stale'. I think that 'stale' is supposed to be "used to be in the Validator set", so "if (this is the Validator network) AND (a node is not a validator) { close connection to it }" but actually this logic seems to close most of the connections most of the time.